### ISSUE TITLE
Extract all service ports when parsing server output

### ISSUE THREAD

**Issue by chanseokoh** [GoogleCloudPlatform/google-cloud-eclipse#945]
~~#938 extracts service port from server output by checking the following line:~~

~~`Starting module "default" running at: `~~

~~[We are embedding `"default"`.](https://github.com/GoogleCloudPlatform/google-cloud-eclipse/pull/938#discussion_r86483644) However, users can configure module names in appengine-web.xml with `<service>`, so this solution is not perfect.~~
UPDATE: #938 will fix the original problem described in this issue.

Note that multi-module runs can show multiple lines of URLs like below, so care should be taken when extracting ports.

```
INFO     2016-11-04 03:16:39,769 devappserver2.py:769] Skipping SDK update check.
INFO     2016-11-04 03:16:39,875 api_server.py:205] Starting API server at: http://localhost:61831
INFO     2016-11-04 03:16:39,882 dispatcher.py:197] Starting module "default" running at: http://localhost:8080
INFO     2016-11-04 03:16:39,886 dispatcher.py:197] Starting module "second" running at: http://localhost:8081
INFO     2016-11-04 03:16:39,888 admin_server.py:116] Starting admin server at: http://localhost:8000
Nov 04, 2016 3:16:41 AM com.google.appengine.tools.development.AbstractContainerService configure
```

**Comment by chanseokoh** [comment]
I propose

1. If the output has a port for `"default"`, set it as the service port.
2. If there is no module with the name `default`, pick a port of an arbitrary module. To determine if there is no `default` module, we have to rely on a further assumption, e.g., the `Starting module ...` output lines will be terminated by `Starting admin server at: `.

**Comment by briandealwis** [comment]
As #916 requests that we implement `IUrlProvider` so as to provide a mapping between a (WTP) module  and its accessible URL, can I suggest that we change the scanner to pull out and track all services URLs. `getServerPort()` could check this list to see if there is  `default` service, and otherwise use the first service?

**Comment by chanseokoh** [comment]
> As #916 requests that we implement `IUrlProvider` so as to provide a mapping between a (WTP) module and its accessible URL, can I suggest that we change the scanner to pull out and track all services URLs.

Good idea, and I guess we should do it to fix #916 anyway.

> `getServerPort() could check this list to see if there is default service, and otherwise use the first service?`

One thing I'd like to point out here is that, at the current state, `getServerPort()` may return a port that is not the port of a `default` module. `getServerPort()` returns the port that the user set  through the UI. In the multi-module scenario, the `default` module may not use the user-specified port.

**Comment by briandealwis** [comment]
Fixed by #1329
